flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/ContextResolvedTableJsonDeserializer.java

```diff
@@ -0,0 +1,220 @@
+        return planRestoreOption != CatalogPlanRestore.ALL_ENFORCED;
+    }
+
+    static ValidationException missingIdentifier() {
+        return new ValidationException(
+                String.format(
+                        "The ContextResolvedTable cannot be deserialized, as no identifier is present within the json, "
+                                + "but lookup is forced by '%s' == '%s'. "
+                                + "Either allow restoring table from the catalog with '%s' == '%s' | '%s' or make sure you don't use anonymous tables when generating the plan.",
+                        TableConfigOptions.PLAN_RESTORE_CATALOG_OBJECTS.key(),
+                        CatalogPlanRestore.IDENTIFIER.name(),
+                        TableConfigOptions.PLAN_RESTORE_CATALOG_OBJECTS.key(),
+                        CatalogPlanRestore.ALL.name(),
+                        CatalogPlanRestore.ALL_ENFORCED.name()));
+    }
+
+    static ValidationException lookupDisabled(ObjectIdentifier objectIdentifier) {
+        return new ValidationException(
+                String.format(
+                        "The ContextResolvedTable with identifier %s does not contain any %s field, "
+                                + "but lookup is disabled because option '%s' == '%s'. "
+                                + "Either enable the catalog lookup with '%s' == '%s' | '%s' or regenerate the plan with '%s' != '%s'.",
+                        objectIdentifier,
+                        FIELD_NAME_CATALOG_TABLE,
+                        TableConfigOptions.PLAN_RESTORE_CATALOG_OBJECTS.key(),
+                        CatalogPlanRestore.ALL_ENFORCED.name(),
+                        TableConfigOptions.PLAN_RESTORE_CATALOG_OBJECTS.key(),
+                        CatalogPlanRestore.IDENTIFIER.name(),
+                        CatalogPlanRestore.ALL.name(),
+                        TableConfigOptions.PLAN_COMPILE_CATALOG_OBJECTS.key(),
+                        TableConfigOptions.CatalogPlanCompilation.IDENTIFIER.name()));
+    }
+
+    static ValidationException schemaNotMatching(
+            ObjectIdentifier objectIdentifier,
+            ResolvedSchema schemaFromPlan,
+            ResolvedSchema schemaFromCatalog) {
+        return new ValidationException(
+                String.format(
+                        "The schema of the table '%s' from the persisted plan does not match the schema loaded from the catalog: '%s' != '%s'. "
+                                + "Have you modified the table schema in the catalog before restoring the plan?.",
+                        objectIdentifier, schemaFromPlan, schemaFromCatalog));
+    }
+
+    static ValidationException missingTableFromCatalog(ObjectIdentifier objectIdentifier) {
+        return new ValidationException(
+                String.format(
+                        "CatalogManager cannot resolve the table with identifier %s and ContextResolvedTable does not contain any %s field. "
```

Comment:
Don't expose internal classes such as `CatalogManager` or `ContextResolvedTable` in exceptions. Also above.